server/users.ts

```diff
@@ -1040,12 +1040,23 @@ export class User extends Chat.MessageContext {
 			Rooms.get(roomid)!.onLeave(oldUser);
 		}
 
-		if (this.locked === '#dnsbl' && !oldUser.locked) this.locked = null;
-		if (!this.locked && oldUser.locked === '#dnsbl') oldUser.locked = null;
-		if (oldUser.locked) this.locked = oldUser.locked;
+		const oldLocked = this.locked;
+
+		if (this.locked && this.locked.startsWith('#') && !oldUser.locked) this.locked = null;
+		if (!this.locked && oldUser.locked && oldUser.locked.startsWith('#')) oldUser.locked = null;
```

Comment:
Wait, what happened to doing this with semilocks, too? Did I miscommunicate in a previous review?

Reply:
Oops, I misread the line that was being deleted.  Fixing now.